src/Containers/Microsoft.NET.Build.Containers/Exceptions/UnableToAccessRepositoryException.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.NET.Build.Containers;
+
+internal sealed class UnableToAccessRepositoryException : Exception
+{
+    public UnableToAccessRepositoryException(string registry)
+        : base(string.Format("Unable to access the registry '{0}'. Please confirm your credentials are correct and that you have access to this registry.", registry))
+    {
+    }
+    public UnableToAccessRepositoryException(string registry, string repository)
+        : base(string.Format("Unable to access the repository '{0}' in the registry '{1}'. Please confirm your credentials are correct and that you have access to this repository and registry.", repository, registry))
```

Comment:
HTTP error useful to surface in this message?

Reply:
I don't think so - it's always stemming from a `401 Unauthorized`, and none of the repositories I tested with included anything meaningful in the response bodies for this error code. Cloud providers are stingy with auth failure details in bodies - they want you to go to their logging infrastructures to learn more to prevent leaking anything.